read_test.go

```diff
@@ -98,3 +104,94 @@ func TestReadWriteNames(t *testing.T) {
 		t.Errorf("got trigram bcd at bits %v, want sz 2", data.fileNameNgrams)
 	}
 }
+
+func loadShard(fn string) (Searcher, error) {
+	f, err := os.Open(fn)
+	if err != nil {
+		return nil, err
+	}
+
+	iFile, err := NewIndexFile(f)
+	if err != nil {
+		return nil, err
+	}
+	s, err := NewSearcher(iFile)
+	if err != nil {
+		iFile.Close()
+		return nil, fmt.Errorf("NewSearcher(%s): %v", fn, err)
+	}
+
+	return s, nil
+}
+
+func TestReadSearch(t *testing.T) {
+	var want []struct {
+		FormatVersion  int
+		FeatureVersion int
+		FileMatches    [][]FileMatch
+	}
+
+	golden := "testdata/golden/TestReadSearch.golden"
+	f, err := os.Open(golden)
+	if err != nil {
+		t.Fatalf("error opening golden file %s", golden)
+	}
+	defer f.Close()
+
+	if err := json.NewDecoder(f).Decode(&want); err != nil {
```

Comment:
don't we want a different golden file per shard? ie I'd suspect we should test shards without symbols.